elytron/WFCORE-4484-ssh-authentication-git-persistence.adoc

```diff
@@ -0,0 +1,136 @@
+** `--algorithm` - the algorithm used to generate the keys
+** `--alias` - the alias to store the KeyPair under
+
+* `--export-public-key` - this will return a single Base64 value which is the public key in it’s encoded form. It will
+have the following options:
+** `--alias` - the alias the KeyPair is stored under
+
+* `--export-private-key` - this will return a single Base64 value which is the private key in it’s encoded form. It will
+have the following options:
+** `--alias` - the alias the KeyPair is stored under
+
+==== Elytron XML File
+
+* It will be possible to specify SSH credentials to be used for authentication in the Elytron XML configuration file
+** There will be the option to specify the existing `key-pair` credential type but with the added option of specifying
+keys in the new OpenSSH format
+** There will be the option to reference a KeyPairCredential stored in a CredentialStore
+** It will be possible to specify the location and identity of a private key (eg. ~/.ssh/id_rsa). The user will also
+be able to specify the passphrase used to encode the key.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
+
+==== JGit/SSHD
+
+* Tests will be added to the WildFly-Core testsuite to test SSH authentication when connecting to a Git repository
+** Tests will be added to test successful authentication when specifying the location, name, and passphrase to the file
+containing a private key
+** Tests will be added to test successful authentication using a KeyPairCredential with all supported algorithm types
+
+==== Elytron Tool
+
+* Tests will be added to the Elytron testsuite to test generating and importing a KeyPairCredential succesfully into
+ a CredentialStore with all supported algorithms
+
+==== Elytron XML File
+* Tests will be added to the Elytron testsuite to test parsing an SSH credential specified as a location, name and passphrase
+of a file
+* Tests will be added to test parsing of a `key-pair` credential type in the OpenSSH format for all supported algorithms
+
+== Community Documentation
+
+* Documentation will be added to https://github.com/wildfly/wildfly/blob/master/docs/src/main/asciidoc/_admin-guide/management-tasks/Configuration_file_git_history.adoc[Git Configuration File History]
+under a new section SSH Authentication detailing how to configure SSH authentication with the different possible credential
+configurations.
+* Documentation will be added to https://github.com/wildfly/wildfly/blob/master/docs/src/main/asciidoc/_elytron/Credential_Store.adoc[Credential Store]
+to describe the new options to generate and import KeyPairCredentials
+* Documentation will be added to https://github.com/wildfly/wildfly/blob/master/docs/src/main/asciidoc/_elytron/Client_Authentication_with_Elytron_Client.adoc[Client Authentication with Elytron Client]
+to describe the changes to the KeyPairCredential and the new option to specify a private keys location as a credential.
```

Comment:
It would be good to add a "Release Notes" section at the end here. This was recently added to the proposal template. See https://lists.jboss.org/pipermail/wildfly-dev/2019-October/007018.html for more details.

Reply:
Added